packages/teams-js/src/internal/communication.ts

```diff
@@ -408,22 +421,34 @@ export function sendMessageToParent(actionName: string, argsOrCallback?: any[] |
   }
 }
 
-const sendMessageToParentHelperLogger = communicationLogger.extend('sendMessageToParentHelper');
+const sendNestedAuthRequestToTopWindowLogger = communicationLogger.extend('sendNestedAuthRequestToTopWindow');
 
 /**
  * @internal
  * Limited to Microsoft-internal use
  */
-function sendMessageToParentHelper(
-  apiVersionTag: string,
-  actionName: string,
-  args: any[] | undefined,
-): MessageRequestWithRequiredProperties {
-  const logger = sendMessageToParentHelperLogger;
-  const targetWindow = Communication.parentWindow;
-  const request = createMessageRequest(apiVersionTag, actionName, args);
+export function sendNestedAuthRequestToTopWindow(message: string): NestedAppAuthRequest {
+  const logger = sendNestedAuthRequestToTopWindowLogger;
 
-  logger('Message %i information: %o', request.id, { actionName, args });
+  const targetWindow = Communication.topWindow;
+  const request = createNestedAppAuthRequest(message);
+
+  logger('Message %i information: %o', request.id, { actionName: request.func });
+
+  return sendRequestToTargetWindowHelper(targetWindow, request) as NestedAppAuthRequest;
+}
+
+const sendRequestToTargetWindowHelperLogger = communicationLogger.extend('sendRequestToTargetWindowHelper');
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function sendRequestToTargetWindowHelper(
+  targetWindow: Window,
+  request: MessageRequestWithRequiredProperties | NestedAppAuthRequest,
+): MessageRequestWithRequiredProperties | NestedAppAuthRequest {
+  const logger = sendRequestToTargetWindowHelperLogger;
 
   if (GlobalVars.isFramelessWindow) {
     if (Communication.currentWindow && Communication.currentWindow.nativeInterface) {
```

Comment:
Some of the logging in this function says it's for parent, but that may or may not be true now that we're using it for parent and sometimes top. Would be good to update those messages